sstables/sstable_set.cc

```diff
@@ -768,6 +768,393 @@ sstable_set_impl::create_single_key_sstable_reader(
     return make_combined_reader(schema, std::move(permit), std::move(readers), fwd, fwd_mr);
 }
 
+// Merges output from readers of TWCS sstables opened for a single partition query into a non-decreasing stream
+// of mutation fragments.
+//
+// Uses `min_position_reader_queue` to open the readers lazily as the read progresses through the partition.
+// A reader is opened only if we find that it may contain fragments for the currently inspected positions.
+//
+// Readers are closed as soon as we find that they were exhausted for the given partition query.
+//
+// Implements the `FragmentProducer` concept. However, `next_partition` and `fast_forward_to(partition_range)`
+// are not implemented and throw an error; the reader is only used for single partition queries.
+//
+// Assumes that the readers returned from `min_position_reader_queue` return fragments from the same partition.
+// Otherwise the behavior is unspecified.
+class time_series_reader_merger {
```

Comment:
A more generic name would be clustering_order_reader_merger. The clustering key doesn't have to be time (although, usually it will be, and the compaction strategy is names Time Window).

Reply:
renamed